### PR TITLE
Phase Space Layout

### PR DESCRIPTION
**Pull request by szabo137**
This PR adds the phase space layout interface and a convenient user-faced API for working with coordinate maps for given processes, models, and phase space layouts. 

# TODO: 

- [x] clean up (remove the TODOs in the files)
- [x] unify the calls of coordinate maps (are the different implementations for in- and out-PSL vital?)
- [x] add validity checks to the interface and construction of the coordinate maps
- [x] fix the docs
- [x] move `patch_QEDbase.jl` to QEDbase

### CODE REVIEW

File: tmp.jl
Comment (AntonReinhard) [inline tmp.jl:1]: Don't forget to remove this again

**Review by AntonReinhard — COMMENTED** [review]
LGTM

**Comment by szabo137** [comment]
@AntonReinhard Is there anything to do with this? If not, I would like to proceed 🙆‍♂️

**Review by AntonReinhard — APPROVED** [review]
All good from my side

**Pull request merged**